packages/hash/api/src/model/knowledge/link.model.ts

```diff
@@ -0,0 +1,160 @@
+   * @param params.targetEntity the target entity of the link
+   */
+  static async create(
+    graphApi: GraphApi,
+    {
+      accountId,
+      sourceEntity,
+      linkTypeModel,
+      targetEntity,
+    }: LinkModelCreateParams,
+  ): Promise<LinkModel> {
+    const {
+      data: { sourceEntityId, linkTypeUri, targetEntityId },
+    } = await graphApi.createLink(sourceEntity.entityId, {
+      accountId,
+      linkTypeUri: linkTypeModel.schema.$id,
+      targetEntityId: targetEntity.entityId,
+    });
+
+    const fetchedParams = {
+      sourceEntity: await EntityModel.getLatest(graphApi, {
+        accountId,
+        entityId: sourceEntityId,
+      }),
+      linkTypeModel: await LinkTypeModel.get(graphApi, {
+        versionedUri: linkTypeUri,
+      }),
+      targetEntity: await EntityModel.getLatest(graphApi, {
+        accountId,
+        entityId: targetEntityId,
+      }),
+    };
+
+    return new LinkModel({
+      accountId,
+      ...fetchedParams,
+    });
+  }
+
+  /**
+   * Get all outgoing links of a source entity.
+   *
+   * @param params.accountId the accountId of the account creating the link
+   * @param params.sourceEntity the source entity of the link
+   */
+  static async getAllOutgoing(
+    graphApi: GraphApi,
+    { sourceEntity }: { sourceEntity: EntityModel },
+  ): Promise<LinkModel[]> {
+    const { data: entityLinks } = await graphApi.getEntityLinks(
+      sourceEntity.entityId,
+    );
+    return Promise.all(
+      Object.entries(entityLinks.outgoing)
+        /**
+         * When target entity id is a list the link has many targets.
+         * This is not supported yet
+         * @todo support 1:N links
+         * */
+        .filter(([_, targetEntityId]) => typeof targetEntityId === "string")
+        .map(
+          async ([linkTypeUri, targetEntityId]) =>
+            new LinkModel({
+              accountId: sourceEntity.accountId,
+              linkTypeModel: await LinkTypeModel.get(graphApi, {
+                versionedUri: linkTypeUri,
+              }),
+              sourceEntity,
+              targetEntity: await EntityModel.getLatest(graphApi, {
+                /** @todo figure out what account ID we use here */
+                accountId: sourceEntity.accountId,
+                entityId: targetEntityId,
+              }),
+            }),
+        ),
+    );
+  }
+
+  /**
+   * Get the outgoing link of a source entity given a link type.
+   *
+   * @param params.accountId the accountId of the account creating the link
+   * @param params.sourceEntity the source entity of the link
+   * @param params.linkTypeModel the Link Type of the link
+   */
+  static async getOutgoing(
+    graphApi: GraphApi,
+    {
+      sourceEntity,
+      linkTypeModel,
+    }: { sourceEntity: EntityModel; linkTypeModel: LinkTypeModel },
+  ): Promise<LinkModel | null> {
+    /** @todo use structural querying for this client-side fetch */
```

Comment:
Similar comment about attaching this TODO to something

Reply:
Addressed in 6e2460441de91db3197537fdc273e40689f83cee